### PR TITLE
add cmake guide docs

### PR DESCRIPTION
**Pull request by drelaptop**
1. improve cmake docs, add using the prebuilt libs docs
2. change the default iOS build target from i386 to 64, make default value alway build success

### CODE REVIEW

**Comment by drelaptop** [comment]
@minggo please review and merge

File: cmake/README.md [hunk 1]
```diff
@@ -1,23 +1,101 @@
...
-* Documentation: [cmake.org/documentation](https://cmake.org/documentation/)
+2. You should use out-of-source build, that is said you need to create a different directory from `cocos2d-x` to execute cmake command.
+
+## Step by Step
+
+### Build in Linux
 
-* CMake FAQ: [Wiki/CMake_FAQ](https://cmake.org/Wiki/CMake_FAQ)
+```sh
+cd cocos2d-x
```
Comment (crazyhappygame) [inline cmake/README.md:17]: why we need for each platform Linux/Win32/Mac different instruction?
It should be everywhere the same:
$ cd cocos2d-x
$ mkdir build
$ cd build
$ cmake ..
$ cmake --build .

File: cmake/README.md [hunk 2]
```diff
@@ -1,23 +1,101 @@
...
+cocos new -l cpp -p my.pack.app2 test_app2
+mkdir app2_build && cd app2_build
+cmake ../test_app2 -DUSE_COCOS_PREBUILT=ON
+make TemplateCpp
+open bin/TemplateCpp.app
+```
+> Any other cpp project can use prebuilt in this way
 
-2. You should use out-of-source build, that is said you need to create a different directory from `$cocos2dx_root` to execute cmake command.
+Using this feature on Android exists a little difference, for CMake can't find system environment when build in Gradle. So you need to [supply a path](https://github.com/cocos2d/cocos2d-x/blob/c087be314c2c56a757bf66163b173746b5d6ad34/tests/cpp-empty-test/proj.android/app/build.gradle#L34) as the location of prebuilt libs.
```
Comment (crazyhappygame) [inline cmake/README.md:98]: It seems like there is access to env from gradle System.getenv("KEYSTORE")
https://stackoverflow.com/questions/25589052/proper-way-to-use-system-environment-variables-in-gradle-using-android-studio

File: cmake/README.md (hunk 1 above)
Comment (drelaptop) [reply cmake/README.md:17]: Right, we can do this in Linux/Win32/Mac native build, not include iOS and Android, different instruction tell users other informations, for example how to generate Xcode project.

Specially, cmake build only support MSVC, didn't support MinGw for win32 platform, so I think adding visible option will avoid conflicts.

>  `cocos2d-x/build` have been used for the older build files, so we can't write `cd cocos2d-x && mkdir build` into the instruction.

File: cmake/README.md (hunk 2 above)
Comment (drelaptop) [reply cmake/README.md:98]: thanks, I will record this idea into https://github.com/cocos2d/cocos2d-x/issues/18717, might improve it in the future.

**Pull request merged**